### PR TITLE
Adds docs for the b-parasite soil moisture sensor

### PR DESCRIPTION
**Pull request by rbaron**
## Description:
[b-parasite](https://github.com/rbaron/b-parasite) is an open hardware and open source soil moisture and ambient temperature/humidity sensor. It broadcasts its sensors' readings via Bluetooth Low Energy (BLE).

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1666

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/b_parasite.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        moisture:
+          name: 'b-parasite Soil Moisture'
+        battery_voltage:
+          name: 'b-parasite Battery Voltage'
+
+Configuration variables
+-----------------------
+
+- **mac_address** (**Required**): The MAC address of the device.
+- **temperature** (**Optional**): Air temperature in Celcius.
```
Comment (OttoWinter) [inline components/sensor/b_parasite.rst]:
```suggestion
- **temperature** (*Optional*): Air temperature in Celcius.
```

We only use bold for `Required` (if it's a different way somewhere else then that's wrong)

File: components/sensor/b_parasite.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  - **name** (**Required**): Sensor name.
```
Comment (OttoWinter) [inline components/sensor/b_parasite.rst:43]: Please add the standard line `- All other options from :ref:`Sensor <config-sensor>`.` here too (same for other sensor blocks)

Comment (rbaron) [reply components/sensor/b_parasite.rst:43]: Done. Thanks!

File: components/sensor/b_parasite.rst (hunk 1 above)
Comment (rbaron) [reply components/sensor/b_parasite.rst]: Done.

**Comment by rbaron** [comment]
Thanks once again @OttoWinter!

**Pull request merged**